### ISSUE TITLE
Boot isn't silent

### ISSUE THREAD

**Issue by nando-digital** [anatol/booster#255]
Booster asks for a password (`Enter passphrase for root_decrypted:`) even though I set the kernel parameter to `silent` and I unlock via TPM. Is there a way to make boot truly silent? Somehow it worked in the past but after I re-setup the boot process (because Windows messed up secure boot)  it didn't work anymore.

**Comment by thePrivacyFanatic** [comment]
[the manpage explains why it works this way](https://github.com/anatol/booster/blob/master/docs/manpage.md#luks-unlock-concurrency-and-prompt-order), so maybe adding an option to give PIN tokens a timeout before appearing so that the race can be won by the others before?

maybe [crypttab](https://github.com/anatol/booster/blob/master/docs/manpage.md#crypttab) can be used here?

also if memory serves there was once an issue where plymouth would swallow the password prompt. if this is still an issue it could be used to bypass the problem (though it should not be enabled on the fallback image to not get locked out)

**Comment by anatol** [comment]
Cc @pilotstew to see if this issue resolved now

**Comment by pilotstew** [comment]
This is a known gap in UX and I've taken a few steps toward a cleaner unlock. As stated in the manpage:
(Assuming your using the recent commit)

> Cancel-on-win. Any prompt — keyboard passphrase, FIDO2-PIN, or TPM2-PIN — is dismissed automatically when a parallel unlock path wins (a touchless token completes, a sibling volume's keyfile reads successfully, etc.). This applies on both the kernel console and the Plymouth splash. On Plymouth the on-screen prompt clears immediately on the server side once Plymouth MR !393 is picked up; older Plymouth builds release the prompt server-side but leave its UI drawn until the splash is otherwise cleared.

For console switching to raw byte read gave us prompt control to cancel/redraw prompts. 

For Plymouth switching to socket gave us a bit more control but ultimately despite great effort I could not resolve prompt dismissal without an upstream [patch](https://gitlab.freedesktop.org/plymouth/plymouth/-/merge_requests/393). I hoped it would have been merged but it hasn't seen much activity so far. This patch would close the Plymouth print as soon as it unlocks via concurrent token like console mode. In the interim the best I could do was offer a status message to notify the Plymouth prompt is dead but can be dismissed by the user (input is swallowed). ..it's possible upvotes and comments could push this along faster. You could manually apply my patch.

So far booster has always raced for unlock, I introduced serial keyboard operations recently. With that work we have most of the tooling in place to offer a flag along the lines of token_priority: strict. Which could force booster into a strict token priority processing rather than concurrent. Individual token timeouts is possible but not done yet and could be a logical extension to strict token priority.

**Comment by pilotstew** [comment]
For the record I think the OP's issue was Windows re-doing secure boot, which changes the PCR values the TPM2 policy was sealed against. Once the policy no longer matches the unseal fails and booster correctly falls back to passphrase. Re-enrolling the keyslot against the current PCRs fixes that. Also worth noting `quiet` just lowers log verbosity and `silent` is only treated as a mount option, neither has anything to do with whether booster prompts. I think this is separate from what @thePrivacyFanatic is raising.

On the concurrency side, one correction to my earlier comment: booster already defers the keyboard passphrase prompt behind tokens. It doesn't show until the token attempts finish or token-timeout elapses (30s default), so a working touchless token like TPM2 PCR-only already boots silent with no prompt drawn. What isn't deferred is the interactive PIN prompt (TPM2-PIN/FIDO2-PIN). That one is drawn as soon as the token dispatches, and cancel-on-win only clears it after the fact (cleanly on console, and on Plymouth once MR !393 lands). 

So what @thePrivacyFanatic is actually after, a delay before PIN prompts so a parallel non-interactive token can win first, is the real gap and I think it has merit. crypttab's token-timeout doesn't cover it. That's a single per-device recovery timer covering all of that device's tokens, not a per-token pre-prompt delay. 

Individual per-token timeouts would be an extension of that logic and a meaningful improvement booster could offer, but would have to be handled in booster's config. 

There's also room for a broader opt-out that processes a device's tokens sequentially and only falls back to the passphrase last, closer to how systemd-cryptsetup behaves, with booster's concurrent unlock staying the default.
